### DISCUSSION TITLE
Sync multiple directories outside the root folder(sync_dir)

### DISCUSSION THREAD

**Discussion by kayzzen01** [abraunegg/onedrive#2649] [Q&A] [answered]
Hi,

I would like to configure/sync multiple directories outside the root folder(sync_dir) in the directories of my home folder

Example:
```
OneDrive: Documents to /home/kayzzen01/Documents
OneDrive: Pictures to /home/kayzzen01/Pictures
```

However, OneDrive Client for Linux only allows this for one directory, using the command:
`onedrive --synchronize --single-directory '<dir_name>'`

Please correct me if I'm wrong.

I don't want to create symbolic links from sync_dir to my home folder. So I'm going to do the opposite, create links from my home folder to the folder where the sync_dir is defined. Example:

```
/home/kayzzen01/Documents symbolic link -> /home/kayzzen01/OneDrive/Documents
/home/kayzzen01/Picture symbolic link -> /home/kayzzen01/OneDrive/Pictures
```

I haven't had time to test it yet, but I don't think it will be a problem. The steps I intend to follow to configure it this way are:

1. Create the `/home/kayzzen/Onedrive` directory
2. Create the links to the Onedrive folder
3. Configure the Onedrive client, sync_dir and the directories I want to sync.

I would like to know if this is the best strategy or if there is another way to do it?

Thanks!

**Comment by abraunegg** [comment, marked as answer]
@kayzzen01 
Microsoft OneDrive has zero concept of symbolic links, so which ever way you choose to surface your directories in your 'sync_dir' these will be uploaded as normal directories.

Personally, I would store all my data in in `/home/kayzzen01/Documents` and the symbolically link `/home/kayzzen01/OneDrive/Documents --> /home/kayzzen01/Documents` ... however this is not what you want to do.

Test with `--dry-run` any setup before putting it into use.

**Comment by kayzzen01** [comment]
It worked perfectly.
First, I made sure everything was in sync, as a --resync is required after any changes made to the settings.

To stay registered and be useful to someone else, here are the steps:

1. It's always good to have a backup! Make sure everything is synchronized, this can be checked with the command:

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`journalctl --user-unit=onedrive -f`

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;It's a good idea to wait about 10 minutes and check if you have the message:

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`Mar 02 15:42:38 arch onedrive[6604]: Sync with OneDrive is complete`

2) Stop the service:

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`systemctl --user stop onedrive`

3) Create the symbolic link:

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`ln -s /home/user/example_dir /home/user/Onedrive/`

4) Configure the sync_list file:

```
# Include OneDrive Root Folders
/old_dir
/example_dir
```
5) Run OneDrive with the --resync option. When the terminal becomes free for command, it means it is finished.

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`onedrive --synchronize --resync`

6) Start the service again in user mode:

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`systemctl --user start onedrive`

Ready!

Thanks @abraunegg
